app/src/main/res/navigation/mobile_navigation.xml

```diff
@@ -49,7 +49,9 @@
         tools:layout="@layout/fragment_closet" >
         <action
             android:id="@+id/action_navigation_closet_to_itemDetailsFragment"
-            app:destination="@id/itemDetailsFragment" />
+            app:destination="@id/itemDetailsFragment"
+            app:popUpTo="@id/action_navigation_closet_to_itemDetailsFragment"
+            app:popUpToInclusive="true"/>
```

Comment:
Change may be unnecessary here. Otherwise LGTM.

Reply:
It's necessary cause I think the back stack wouldn't work